acme_diags/driver/utils/diurnal_cycle.py

```diff
@@ -56,11 +56,13 @@ def composite_diurnal_cycle(var, season):
 
     ncycle = len(cycle)
     for n in range(ncycle):
-        idx = numpy.array( [ season_idx[cycle[n]][var_time_absolute[i].month-1]
-                          for i in range(len(var_time_absolute)) ], dtype=numpy.int).nonzero()
-
+        # Get time index for each month/season. 
+        idx = numpy.array([season_idx[cycle[n]][var_time_absolute[i].month-1]
+                          for i in range(len(var_time_absolute))], dtype=numpy.int).nonzero()
+        # var_season has shape (ncycle, ntimesteps, [lat,lon]) 
```

Comment:
I'm a bit confused as to how `len(idx[0])` is `ntimesteps`. Say the first season is `'DJF'`. If `n == 0`, then `cycle[n] == cycle[0] == 'DJF'`. Then, `season_idx[cycle[n]] == season_index['DJF'] == [1,1,0,0,0,0,0,0,0,0,0,1]`. Initially, the list comprehension has `i == 0`, so if `var_time_absolute[0].month == '2'` for  February, then we would take `2-1=1` as the index. That gives us `idx[0] == season_idx['DJF'][1] == 1`. It seems to me like `ntimesteps` is actually equal to `len(idx)` then. Am I not following correctly? Thanks.

Reply:
Yes it is a bit confusing. `nonzero()` return a tuple of array, so len(idx[0]) gives the actual `ntimesteps`. i.e., len(idx)= 1 (only one element in that tuple), len(idx[0]) equals to 8, which is the number of time steps, given climatology diurnal cycle of one season.

Reply:
Oh, that makes sense. Thank you!

Reply:
Thank you for the careful review.